sys/vm/vm_kern.c

```diff
@@ -633,21 +635,34 @@ kmem_free(vm_offset_t addr, vm_size_t size)
 vm_offset_t
 kmap_alloc_wait(vm_map_t map, vm_size_t size)
 {
+	int error;
+	vm_size_t padded_size;
+	vm_offset_t alignment;
 	vm_offset_t addr;
+	vm_pointer_t mapped;
 
 	size = round_page(size);
 	if (!swap_reserve(size))
 		return (0);
+	padded_size = CHERI_REPRESENTABLE_LENGTH(size);
+	alignment = CHERI_REPRESENTABLE_ALIGNMENT(size);
 
 	for (;;) {
 		/*
 		 * To make this work for more than one map, use the map's lock
 		 * to lock out sleepers/wakers.
 		 */
 		vm_map_lock(map);
-		addr = vm_map_findspace(map, vm_map_min(map), size);
-		if (addr + size <= vm_map_max(map))
+		addr = vm_map_findspace(map, vm_map_min(map), padded_size);
+		if (addr + padded_size <= vm_map_max(map) && alignment == 0)
 			break;
+		if (alignment > 0) {
+			error = vm_map_alignspace(map, NULL, 0, &addr,
```

Comment:
It seems like `vm_map_findspace` should be doing this? Then you don't need to expose the function.

Reply:
I think I can make `vm_map_findspace` always ensure representability. I'm unsure about how this would play with vm_maps with mixed modes, e.g. when used to allocate from hybrid-mode we do not require/expect to increase the alignment requirements, but I think I can conditionally do that, depending on the MAP_RESERVATIONS flag.

Reply:
We have decided to keep `vm_map_alignspace` as changing the behaviour of `vm_map_findspace` would require a function signature change as findspace would be allowed to pad the allocation size and return the increased length. This is more disruptive to the diff.